src/control/cam_comp.F90

```diff
@@ -536,9 +580,25 @@ subroutine cam_register_constituents(cam_runtime_opts)
       ! Grab a pointer to the constituent array
       const_props => cam_model_const_properties()
 
-      ! Finally, initialize the constituents module
+      ! Initialize the constituents module
       call cam_constituents_init(const_props, num_advect)
 
+      ! Finally, for CAM moist physics, one will need to use a
+      ! non-zero minimum value for water vapor, so update that
+      ! value here:
+      !-------------------------------------------
+      if (phys_suite_name /= 'held_suarez_1994') then !Held-Suarez is "dry" physics
```

Comment:
would it make sense to have some sort of is_dry logical for schemes? Is it possible that we'll have other dry physics? Or is this a PFFU (problem for future us)

Reply:
I don't personally think we should worry about adding a logical here, for two reasons:

1.  Only Held-Suarez and "adiabatic" (which basically means no physics at all) have the minimum set to zero.  All other CAM physics schemes use the default minimum value.  So I feel like adding a logical and a framework to support that logical would probably not be worth the effort (at least for CAM physics suites).

2.  Ideally I don't want CAM-SIMA to add water vapor (or any constituent) by default.  Instead it should be up to the physics, chemistry, and dycore to add whatever constituents they need.  This means that eventually I hope this whole code block will go away.  Given this, I want to put the bare minimum needed into this particular set of routines.  As an FYI, there is already an issue open (issue #208) for this future development plan.

Of course if you disagree with either of these points, or think there would be additional benefits to having an `is_dry` logical, then just let me know!

Reply:
Is there a replacement for the `moist_physics` logical that is used, for example, in cam_diagnostics.F90?

Reply:
@gold2718 Not yet, but since we don't have the new history infrastructure plugged in yet I figure we can tackle that particular concern later (but I definitely agree we need some general way of allowing physics suites to tell CAM-SIMA what standard outputs should or should not be included by default).